### ISSUE TITLE
log_every_n_steps = 1, but the logs are logged every 2 steps

### ISSUE THREAD

**Issue by davidkflau** [Lightning-AI/lightning#12643]
I configured log_every_n_steps = 1 in the Trainer(), and log the training loss in the training_step_end. However, I found that the training loss is logged every two steps.

```
trainer = pl.Trainer(
    gpus=2,
    max_epochs=5,
    progress_bar_refresh_rate=20,
    val_check_interval=0.1,
    accelerator="ddp",
    logger=mlf_logger,
    num_nodes=2,
    log_every_n_steps=1
  )
```

```
def training_step_end(self, batch_parts):
    # Generator's loss from GPUs
    g_train_loss_mean = torch.mean(batch_parts["g_loss"])
    self.log('g_train_loss', g_train_loss_mean, on_step=True, on_epoch=False, prog_bar=True, logger=True) 
    # Discriminator's loss from GPUs
    d_train_loss_mean = torch.mean(batch_parts["d_loss"])
    self.log('d_train_loss', d_train_loss_mean, on_step=True, on_epoch=False, prog_bar=True, logger=True)
```

I tried to log the `self.global_step` and found it's logged every two steps.
<img width="503" alt="image" src="https://user-images.githubusercontent.com/46880725/161992165-57ee08e3-9882-4aec-9a23-08e399b11746.png">

cc @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

**Comment by rohitgr7** [comment]
from your code, looks like you are using manual optimization and doing 2 calls to the optimizer inside training_step, one for the generator and another one for the discriminator.

in v1.6 the behavior of global_step was updated. Check out the docs: https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#global-step
